frontend/src/core/comments/components/AddComment.js

```diff
@@ -25,69 +34,87 @@ export default function AddComments(props: Props) {
         addComment,
     } = props;
 
-    const commentInput: any = React.useRef();
-    const minRows = 1;
-    const maxRows = 6;
+    let [editorState, setEditorState] = React.useState(
+        EditorState.createEmpty()
+    )
+    const editor: any = React.useRef(null);
+    const rawContentState = convertToRaw(editorState.getCurrentContent());
 
-    if (!user) {
-        return null;
+    const onChange = (editorState) => {
+        setEditorState(editorState);
     }
 
-    const handleOnChange = () => {
-        const textAreaLineHeight = 24;
-        commentInput.current.rows = minRows;
-
-        const currentRows = Math.trunc(commentInput.current.scrollHeight / textAreaLineHeight);
-
-        if (currentRows < maxRows) {
-            commentInput.current.rows = currentRows;
-        }
-        else {
-            commentInput.current.rows = maxRows;
-        }
+    function focusEditor() {
+        editor.current.focus();
     }
 
-    const handleOnKeyDown = (event: SyntheticKeyboardEvent<>) => {
-        if (event.keyCode === 13 && event.shiftKey === false) {
-            submitComment(event);
+    function keyBindingFn(e: SyntheticKeyboardEvent<>) {
+        if (e.keyCode === 13 && e.shiftKey === false) {
+          return 'submitOnEnter' 
         }
+        
+        return getDefaultKeyBinding(e)
     }
 
-    const submitComment = (event: SyntheticEvent<>) => {
-        event.preventDefault();
-        const comment = commentInput.current.value;
+    React.useEffect(() => {
+        focusEditor()
+    }, []);
 
-        if (!comment) {
-            return null;
+    const handleKeyCommand = (command: string) => {
+        if (command === 'submitOnEnter') {
+            submitComment();
+            return 'handled';
         }
+        return 'not-handled';
+    }
+
+    const clearEditorContent = () => {
+        let contentState = editorState.getCurrentContent();
+        const firstBlock = contentState.getFirstBlock();
+        const lastBlock = contentState.getLastBlock();
+        const allSelected = new SelectionState({
+            anchorKey: firstBlock.getKey(),
+            anchorOffset: 0,
+            focusKey: lastBlock.getKey(),
+            focusOffset: lastBlock.getLength(),
+            hasFocus: true
+        });
+        contentState = Modifier.removeRange(contentState, allSelected, 'backward');
+        editorState = EditorState.push(editorState, contentState, 'remove-range');
+        editorState = EditorState.forceSelection(editorState, contentState.getSelectionAfter());
+        setEditorState(editorState);
+    }
 
+    const submitComment = () => {
+        const comment = draftToHtml(
+            rawContentState,
+        )
+        
         addComment(comment, translation);
 
-        commentInput.current.value = '';
-        commentInput.current.rows = minRows;
-    };
+        clearEditorContent();
+    }
+
+    if (!user) {
+        return null;
+    }
```

Comment:
This one should be somewhere up, so that we don't execute all the code for non-authenticated users.

Reply:
Nit: It can go further up, right after you declare props consts. No need to use state, ref or focus if you don't have a user.

Reply:
Actually, I can't per the "rules of hooks" 😦 

<img width="676" alt="Screen Shot 2020-04-10 at 8 14 01 AM" src="https://user-images.githubusercontent.com/19564687/79005998-fba5ea00-7b0c-11ea-8065-7b5e550f9acf.png">

Reply:
Oh, wow! I had no idea. Sorry. 🤷‍♂️ 😸